src/history.rs

```diff
@@ -654,45 +659,41 @@ impl HistoryImpl {
             wrealpath(&possibly_indirect_target_name).unwrap_or(possibly_indirect_target_name);
 
         // Make our temporary file
-        // Remember that we have to close this fd!
-        let Some((tmp_file, tmp_name)) = create_temporary_file(&tmp_name_template) else {
+        let Some((mut tmp_file, tmp_name)) = create_temporary_file(&tmp_name_template) else {
             return;
         };
-        let tmp_fd = tmp_file.as_raw_fd();
         let mut done = false;
         for _i in 0..MAX_SAVE_TRIES {
             if done {
                 break;
             }
 
-            let target_fd_before = wopen_cloexec(
+            let target_file_before = wopen_cloexec(
                 &target_name,
                 OFlag::O_RDONLY | OFlag::O_CREAT,
                 HISTORY_FILE_MODE,
-            );
+            )
+            .map(|fd| unsafe { File::from_raw_fd(fd.into_raw_fd()) });
```

Comment:
Same case here, the fd is already a safe Owned variant.
```suggestion
            .map(File::from);
```